### PR TITLE
feat: [P4ADEV-166] Implemented footer

### PR DESCRIPTION
**Pull request by AnisLucidi01**
#### List of Changes

- Added and configured footer component from MUI Italia, and adapted languages file for it. 
- Created footer story.
- Added a global react hook which implements language changement functionality. 
- UseLanguage hook unit test
- Added "baseUrl": "src" to tsconfig.json (now is possible to use absolute import considering src as base folder)
- Configurated vite to use tsconfig.json as config file

#### How Has This Been Tested?

Visual testing.

#### Screenshots (if appropriate):
image

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/components/Layout.tsx [hunk 1]
```diff
@@ -30,7 +37,73 @@ export function Layout({ children }: LayoutProps) {
           {children}
         </Grid>
         <Grid item xs={12}>
-          <Footer />
+          <Footer
```
Comment (stratoivandiluccio) [inline src/components/Layout.tsx:40]: Could you restore the Footer.tsx file and write an own Footer component that return the mui Footer component configured like that?

File: src/App.tsx [hunk 2]
```diff
@@ -1,35 +1,33 @@
...
 
 export const App = () => {
   const { t } = useTranslation();
   document.title = t('app.title');
+  const [currentLanguage, setCurrentLanguage] = useState<LangCode>('it');
+
+  const changeLanguage = (langCode: LangCode) => {
+    i18n.changeLanguage(langCode);
+    setCurrentLanguage(langCode);
+  };
```
Comment (stratoivandiluccio) [inline src/App.tsx:24]: Please move this logic in an external hook so you can write some unit test easily

**Review by stratoivandiluccio — CHANGES_REQUESTED** [review]
I added some feedback as comment

File: src/components/Layout.tsx [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
 
-export function Layout({ children }: LayoutProps) {
+export function Layout({ children, currentLanguage, changeLanguage }: LayoutProps) {
+  const { t } = useTranslation();
+
   const theme = useTheme();
   return (
     <Container
-      maxWidth="lg"
+      maxWidth={false}
```
[the remark this answers is not in the document]
Comment (stratoivandiluccio) [reply src/components/Layout.tsx:24]: It's ok for now to have the footer non stretching the whole page width. You should just restore 'lg' and doing nothing else

**Review by mmoio — APPROVED** [review]

**Pull request merged**